woocommerce/woocommerce
# Product Editor Onboarding: Add "tell me more" button to wc.com page

### Submission Review Guidelines:

- I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
- I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
- I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/). 
- Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:

Add "tell me more" button to wc.com page.
 
On top of that, I copied the Guide component from `@wordpress/components`, because I had to implement some things on it, as described in the comment left in the code.

I also slightly tweaked the place where the option `woocommerce_block_product_tour_shown` is set. It is only set when the guide is closed or finished.





Closes #38518 .



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Enable the block product editor in WooCommerce > Settings > Advanced > Features.
2. Go to Product > Add New.
3. You should see the tour kit step in the bottom left of the editor. (If you have already dismissed it, you can delete the `woocommerce_block_product_tour_shown` option through WCA Test Helper)
4. Click 'View highlights'
5. Navigate through the guide
6. Check that the Close button is not focused every time the page switches
7. Close the Guide by clicking the Close button.
8. Verify that the `wcadmin_block_product_editor_spotlight_dismissed` tracks event was fired with the correct `current_page`.
9. Delete the `woocommerce_block_product_tour_shown` through the WCA Test Helper and refresh the page.
10. Go through the Guide again until the end and click 'Tell me more'
11. Verify that a new tab was opened with more information.
